File: app/src/main/java/com/example/unsplash/model/unsplash/HeaderInterceptorPost.java
package com.example.unsplash.model.unsplash;

import android.support.annotation.NonNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class HeaderInterceptorPost implements Interceptor {


    private String token;

    HeaderInterceptorPost(String clientId) {
        this.token = clientId;
    }

    @NonNull
    @Override
    public Response intercept(@NonNull Interceptor.Chain chain) throws IOException {
        Request request = chain.request();
        request = request.newBuilder()
                .addHeader("Authorization", "Bearer " + token)
                .build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/example/unsplash/model/unsplash/HeaderInterceptorPost.kt
package com.example.unsplash.model.unsplash

import java.io.IOException
import okhttp3.Interceptor
import okhttp3.Response

class HeaderInterceptorPost internal constructor(private val token: String) : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var request = chain.request()
        request = request.newBuilder()
                .addHeader("Authorization", "Bearer $token")
                .build()
        return chain.proceed(request)
    }
}
